crates/re_space_view_time_series/src/visualizer_system.rs

```diff
@@ -1,91 +1,34 @@
-use re_data_store::TimeRange;
-use re_log_types::{EntityPath, StoreKind, TimeInt};
 use re_query_cache::{MaybeCachedComponentData, QueryError};
 use re_types::{
     archetypes::TimeSeriesScalar,
     components::{Color, Radius, Scalar, ScalarScattering, Text},
-    Component, Loggable,
+    Loggable,
 };
 use re_viewer_context::{
-    external::re_entity_db::TimeSeriesAggregator, AnnotationMap, DefaultColor,
-    IdentifiedViewSystem, ResolvedAnnotationInfo, SpaceViewSystemExecutionError, ViewQuery,
+    AnnotationMap, DefaultColor, IdentifiedViewSystem, SpaceViewSystemExecutionError, ViewQuery,
     ViewerContext, VisualizerQueryInfo, VisualizerSystem,
 };
 
-// ---
-
-#[derive(Clone, Debug)]
-pub struct PlotPointAttrs {
-    pub label: Option<String>,
-    pub color: egui::Color32,
-    pub radius: f32,
-    pub kind: PlotSeriesKind,
-}
-
-impl PartialEq for PlotPointAttrs {
-    fn eq(&self, rhs: &Self) -> bool {
-        let Self {
-            label,
-            color,
-            radius,
-            kind,
-        } = self;
-        label.eq(&rhs.label)
-            && color.eq(&rhs.color)
-            && radius.total_cmp(&rhs.radius).is_eq()
-            && kind.eq(&rhs.kind)
-    }
-}
-
-impl Eq for PlotPointAttrs {}
-
-#[derive(Clone, Debug)]
-struct PlotPoint {
-    time: i64,
-    value: f64,
-    attrs: PlotPointAttrs,
-}
-
-#[derive(Clone, Copy, Debug, PartialEq, Eq)]
-pub enum PlotSeriesKind {
-    Continuous,
-    Scatter,
-    Clear,
-}
-
-#[derive(Clone, Debug)]
-pub struct PlotSeries {
-    pub label: String,
-    pub color: egui::Color32,
-    pub width: f32,
-    pub kind: PlotSeriesKind,
-    pub points: Vec<(i64, f64)>,
-    pub entity_path: EntityPath,
-}
+use crate::{
+    overrides::{initial_override_color, lookup_override},
+    util::{determine_time_range, points_to_lines},
+    PlotPoint, PlotPointAttrs, PlotSeries, PlotSeriesKind,
+};
 
-/// A scene for a time series plot, with everything needed to render it.
+/// The legacy system for rendering [`TimeSeriesScalars`] archetypes.
 #[derive(Default, Debug)]
-pub struct TimeSeriesSystem {
+pub struct LegacyTimeSeriesSystem {
     pub annotation_map: AnnotationMap,
     pub lines: Vec<PlotSeries>,
-
-    /// Earliest time an entity was recorded at on the current timeline.
-    pub min_time: Option<i64>,
-
-    /// What kind of aggregation was used to compute the graph?
-    pub aggregator: TimeSeriesAggregator,
-
-    /// `1.0` for raw data.
-    pub aggregation_factor: f64,
 }
 
-impl IdentifiedViewSystem for TimeSeriesSystem {
+impl IdentifiedViewSystem for LegacyTimeSeriesSystem {
     fn identifier() -> re_viewer_context::ViewSystemIdentifier {
-        "TimeSeries".into()
+        "LegacyTimeSeries".into()
     }
 }
 
-impl VisualizerSystem for TimeSeriesSystem {
+impl VisualizerSystem for LegacyTimeSeriesSystem {
     fn visualizer_query_info(&self) -> VisualizerQueryInfo {
         VisualizerQueryInfo::from_archetype::<TimeSeriesScalar>()
```

Comment:
we should still leave some note on `TimeSeriesScalar` that it's on the way out. Otherwise, we always end up with the legacy time series system which might be confusing ("why is it saying legacy no matter what I do")